File: src/Python/908.SmallestRangeI.py

class Solution:
    def smallestRangeI(self, A, K):
        """
        :type A: List[int]
        :type K: int
        :rtype: int
        """
        return max(max(A) - min(A) - 2 * K, 0)


if __name__ == '__main__':
    solution = Solution()
    print(solution.smallestRangeI(A = [1], K = 0))
    print(solution.smallestRangeI(A = [0,10], K = 2))
    print(solution.smallestRangeI(A = [1,3,6], K = 3))
    print(solution.smallestRangeI(A = [2,7,2], K = 1))
else:
    pass

File: src/Java/908.SmallestRangeI.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public int smallestRangeI(int[] A, int K) {
        int maxv = 0;
        int minv = 10000;
        for (int a : A){
            maxv = Math.max(a, maxv);
            minv = Math.min(a, minv);
        }
        return Math.max(0, maxv - minv - 2 * K);
    }
}
